src/transformers/models/jukebox/modeling_jukebox.py

```diff
@@ -0,0 +1,3133 @@
+            if self.single_enc_dec:
+                if no_past_context:
+                    music_tokens, audio_conditioning = self.prior_preprocess(
+                        [lyric_tokens], [None, audio_conditioning]
+                    )
+                else:
+                    music_tokens, audio_conditioning = self.prior_preprocess(
+                        [lyric_tokens, music_tokens], [None, audio_conditioning]
+                    )
+                if sample_tokens is not None:
+                    sample_tokens += self.nb_relevant_lyric_tokens
+                tokens = self.prior.primed_sample(
+                    n_samples,
+                    music_tokens,
+                    audio_conditioning,
+                    metadata_conditioning,
+                    fp16=fp16,
+                    temp=temp,
+                    top_k=top_k,
+                    top_p=top_p,
+                    chunk_size=chunk_size,
+                    sample_tokens=sample_tokens,
+                )
+                music_tokens = self.prior_postprocess(tokens)
+            else:
+                lyric_encoder_states = self.get_lyric_encoder_states(lyric_tokens, fp16=fp16, sample=True)
+                if no_past_context:
+                    music_tokens = self.prior.sample(
+                        n_samples,
+                        audio_conditioning,
+                        metadata_conditioning,
+                        lyric_encoder_states,
+                        fp16=fp16,
+                        temp=temp,
+                        top_k=top_k,
+                        top_p=top_p,
+                        sample_tokens=sample_tokens,
+                    )
+                else:
+                    music_tokens = self.prior.primed_sample(
+                        n_samples,
+                        music_tokens,
+                        audio_conditioning,
+                        metadata_conditioning,
+                        lyric_encoder_states,
+                        fp16=fp16,
+                        temp=temp,
+                        top_k=top_k,
+                        top_p=top_p,
+                        chunk_size=chunk_size,
+                        sample_tokens=sample_tokens,
+                    )
+        return music_tokens
+
+    def get_lyric_encoder_states(self, lyric_tokens, fp16=False, sample=False):
+        """
+        Retreive the last hidden_states of the lyric encoder that will be attended to by the decoder. Forwards through
+        the lyric encoder.
+        """
+        if self.nb_relevant_lyric_tokens != 0 and self.lyric_conditioning:
+            if sample:
+                self.lyric_encoder = self.lyric_encoder.to(lyric_tokens.device)
+            lyric_acts = self.lyric_encoder(lyric_tokens, None, None, None, fp16=fp16)
+            lyric_acts = self.lyric_encoder.proj_in(lyric_acts)
+            if fp16:
+                lyric_acts = lyric_acts.half()
```

Comment:
I don't understand this -> if `self.lyric_encoder` is in `half()` => then `lyric_acts` will automatically be in `half()` as well -> we should not  need to pass a `fp16` flag here